packages/taco-auth/src/providers/eip4361/eip4361.ts

```diff
@@ -9,34 +10,73 @@ import {
 
-export class EIP4361AuthProvider {
+/**
+ * EIP4361AuthProvider implements the AuthProvider interface to handle Sign-In with Ethereum (EIP-4361/SIWE) authentication.
+ * This provider manages the creation, storage, and verification of SIWE messages for authentication purposes.
+ */
+export class EIP4361AuthProvider implements AuthProvider {
   private readonly storage: LocalStorage<EIP4361AuthSignature>;
-  private readonly providerParams: EIP4361AuthProviderParams;
-
+  private readonly params: Partial<SiweMessage>;
+  private readonly expirationDuration: number;
+  /**
+   * Creates a new EIP4361AuthProvider to handle Sign-In with Ethereum (SIWE) authentication.
+   *
+   * This provider generates and manages SIWE messages with configurable parameters while enforcing
+   * security best practices like expiration time limits. It automatically handles message storage
+   * and retrieval using LocalStorage.
+   *
+   * @param provider - An ethers provider used to fetch the current chainId.
+   * @param signer - An ethers signer used to sign the SIWE message.
+   * @param params - Optional configuration parameters for the SIWE message.
+   * @param params.domain - Domain name for the signing request. Defaults to current website domain or 'taco.build'.
+   * @param params.uri - URI identifying the signing request origin. Defaults to current website URL or 'https://taco.build'.
+   * @param params.version - SIWE message version. Defaults to '1'.
+   * @param params.nonce - Unique identifier for the signing request. Auto-generated if not provided.
+   * @param params.expirationTime - ISO datetime when the signature expires. Must be within 2 hours from now. The value will be used to infer the expiration duration timespan.
+   * @param params.statement - Human-readable message shown to users during signing. Default is `{{domain}} wants you to sign in with your Ethereum account: {{address}}`.
+   * @throws {Error} If expirationTime is set beyond 2 hours from the current time.
+   */
   constructor(
     // TODO: We only need the provider to fetch the chainId, consider removing it
     private readonly provider: ethers.providers.Provider,
     private readonly signer: ethers.Signer,
-    providerParams?: EIP4361AuthProviderParams,
+    // `chainId` will be set from the provider. For do not accept it as input to eleminate ambiguity.
+    params?: Partial<Omit<Omit<SiweMessage, 'address'>, 'chainId'>>,
   ) {
     this.storage = new LocalStorage(eip4361AuthSignatureSchema);
-    if (providerParams) {
-      this.providerParams = providerParams;
-    } else {
-      this.providerParams = this.getDefaultParameters();
+
+    const after2HoursFromNow = new Date(
+      Date.now() + 2 * 60 * 60 * 1000,
+    ).toISOString();
+    // if the provided expirationTime was more than 2 hours from now, throw
+    if (params?.expirationTime && params.expirationTime > after2HoursFromNow) {
+      throw new Error('expirationTime is too soon');
```

Comment:
Probably I'm misunderstanding this, doesn't the error be "expirationTime is too late"? 

We are providing to the constructor the expiration time for SIWE cache, and we are limiting how long this can be (2h). So, if the user tries to set a expiration time longer than 2 hours (`params.expirationTime > after2HoursFromNow`), the error say that the expiration time is too late. Right?

Reply:
Sorry for the confusion, the message was for something and I missed updating it.